plugin-process-to-cgroup-bridge/src/transform.rs

```
@@ -0,0 +1,199 @@
+    keep_processed_measurements: bool,
+}
+
+impl Transform for ProcessToCgroupBridgeTransform {
+    fn apply(&mut self, measurements: &mut MeasurementBuffer, _ctx: &TransformContext) -> Result<(), TransformError> {
+        let mut cgroup_measurements = MeasurementBuffer::new();
+        let mut kept_measurements = MeasurementBuffer::new();
+        for measurement in measurements.iter() {
+            if let Some(cgroup_measurement) = self.cgroup_measurement_from_measurement(measurement)? {
+                cgroup_measurements.push(cgroup_measurement);
+                if self.keep_processed_measurements {
+                    kept_measurements.push(measurement.clone());
+                }
+            } else {
+                kept_measurements.push(measurement.clone());
+            }
+        }
+        measurements.clear();
+
+        if self.merge_similar_cgroups {
+            let mut aggregated_cgroup_measurements = Self::aggregate_cgroups_measurements(&mut cgroup_measurements);
+            cgroup_measurements.clear();
+            cgroup_measurements.merge(&mut aggregated_cgroup_measurements);
+        }
+
+        measurements.merge(&mut kept_measurements);
+        measurements.merge(&mut cgroup_measurements);
+
+        Ok(())
+    }
+}
+
+impl ProcessToCgroupBridgeTransform {
+    pub fn new(
+        processes_metrics_ids: Vec<RawMetricId>,
+        merge_similar_cgroups: bool,
+        keep_processed_measurements: bool,
+    ) -> Self {
+        Self {
+            processes_metrics_ids,
+            merge_similar_cgroups,
+            keep_processed_measurements,
+        }
+    }
+
+    fn cgroup_measurement_from_measurement(
+        &self,
+        measurement: &MeasurementPoint,
+    ) -> anyhow::Result<Option<MeasurementPoint>> {
+        if !self.processes_metrics_ids.contains(&measurement.metric) {
+            return Ok(None);
+        }
+        let pid = match Self::extract_process_id_from_measurement(measurement) {
```

If you ignore the error of `extract_process_id_from_measurement`, it should probably return `Option` instead.